src/main.rs

```diff
@@ -5,15 +5,23 @@ use std::thread;
 use std::net::{TcpListener, TcpStream, Shutdown};
 use std::io::{Read, Write};
 
-const BUFFER_SIZE: usize = 250;
+pub mod translate;
+pub mod message;
+
+const BUFFER_SIZE: usize = 256;
 const PORT: u16 = 3333;
 
 fn handle_client(mut stream: TcpStream) {
     let mut data = [0 as u8; BUFFER_SIZE];
     while match stream.read(&mut data) {
         Ok(size) => {
             trace!("stream read {} bytes", size);
-            stream.write(&data[0..size]).unwrap();
+            let request = crate::message::Request::deserialize(&data);
```

Comment:
What happens if the message doesn't match the format expected for deserialization?